Detectors/Vertexing/src/SVertexer.cxx

```diff
@@ -513,18 +539,23 @@ int SVertexer::checkCascades(float rv0, std::array<float, 3> pV0, float p2V0, in
       }
     }
     if (!goodHyp) {
+      LOG(debug) << "Casc not compatible with any hypothesis";
       continue;
     }
     auto& casc = mCascadesTmp[ithread].emplace_back(cascXYZ, pCasc, fitterCasc.calcPCACovMatrixFlat(candC), trNeut, trBach, mV0sTmp[ithread].size() - 1, bach.gid);
     o2::track::TrackParCov trc = casc;
     o2::dataformats::DCA dca;
-    if (!trc.propagateToDCA(pv, fitterCasc.getBz(), &dca, 5.) ||
+    if (!trc.propagateToDCA(cascPv, fitterCasc.getBz(), &dca, 5.) ||
         std::abs(dca.getY()) > mSVParams->maxDCAXYCasc || std::abs(dca.getZ()) > mSVParams->maxDCAZCasc) {
+      LOG(debug) << "Casc not compatible with PV";
+      LOG(debug) << "DCA: " << dca.getY() << " " << dca.getZ();
       mCascadesTmp[ithread].pop_back();
       continue;
     }
-    casc.setCosPA(cosPA);
-    casc.setVertexID(v0.getVertexID());
+
+    LOG(debug) << "Casc successfully added";
+    casc.setCosPA(bestCosPA);
+    casc.setVertexID(cascVtxID);
```

Comment:
The cascade contains the V0 ID, e.g. in the AOD it is stored simply as a reference on V0 and bachelor track indices: https://github.com/AliceO2Group/AliceO2/blob/a9d875e3b1082d218489380b536036b1d103b45a/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L972.
With this fix you may have a situation when V0 is registered for one PV while the cascade using this v0 is for a different PV. I think you need to readjust also the PV index and cosPA of the V0. 
But given that you may add different cascades for different PVs using the same V0, I think you will need to replicate the V0 if the PV used by the cascade is not the same as the PV used by the original V0.